aea/decision_maker/base.py

```diff
@@ -301,11 +305,36 @@ def _handle_tx_message(self, tx_message: TransactionMessage) -> None:
         :param tx_message: the transaction message
         :return: None
         """
-        score_diff = self.preferences.get_score_diff_from_transaction(self.ownership_state, tx_message)
-        if score_diff >= 0.0:
-            pass  # TODO: implement tx flow
-            # envelope = Envelope()
-            # self.outbox.put(envelope)
+        amount = tx_message.get("amount")
+        m_address = self.generate_address_from_public_key(self._wallet.public_keys['fetchai'])
+        api = LedgerApi("127.0.0.1", 8100)
+        self._generate_wealth()
+        print(api.tokens.balance(m_address))
+        if amount <= api.tokens.balance(m_address):
+            m_entity = Entity.from_hex(self._wallet.crypto_objects['fetchai'].private_key)
+            to = self.generate_address_from_public_key(tx_message.get("counterparty"))
```

Comment:
We should already pass the `counterpart` address into the tx_message (since the counterpart will have to send it to us)

Reply:
The counterpart sends the public_key and not the actual address. We had this discussion and we said that since we can generate the address from the public key there is no need to change how we store things in the wallet.

Reply:
Ok, so in this case you can just create a helper method which is available from aea.crypto.helpers and lets you recover the address from the public key